### PR TITLE
Vue app cleanup

### PR DESCRIPTION
**Pull request by Mojito-Fivem**
Got rid of the brunt of your bloat, reduced the bundle size by 38% by simply changing some dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +16,7 @@
         "jquery": "^3.6.0",
         "jquery-ui": "^1.13.1",
         "jsdom": "^19.0.0",
```
Comment (restray) [inline package.json:18]: You can delete those lines too (don't need those, I used them for development)

**Review by restray — CHANGES_REQUESTED** [review]
Just need to remove 3 lines in the `package.json` but you did a really good job! Thanks you so much :)

**Comment by Mojito-Fivem** [comment]
I went through and deleted all instances of the `var` keyword, as of ES6 this is deprecated and let/const should be used in it's place. Also fixed some instances of using (==) instead of (===)

**Comment by restray** [comment]
> If you're migrating your code to Vue you should at least consider using typescript here.

I don't agree with you at all... VueJs is already complicated for most of the servers owners! And most of the FiveM resources works with JS and not TS! Those are my arguments to "not moving on TS"

**Review by restray — APPROVED** [review]
It seems pretty good!

**Comment by Mojito-Fivem** [comment]
> > If you're migrating your code to Vue you should at least consider using typescript here.
> 
> I don't agree with you at all... VueJs is already complicated for most of the servers owners! And most of the FiveM resources works with JS and not TS! Those are my arguments to "not moving on TS"

Exactly. If you're already going to put working on this resources out of the hands of 99% of the userbase, why would you not go the extra step and use typescript which gives a much better developer experience for not all that much work, I see you already using jsdocs here (some of them are wrong but this PR doesn't correct them) so why would you not simplify with typescript which will self document these functions

File: html_app/components/AttachmentMenu.vue [hunk 2]
```diff
@@ -183,19 +185,16 @@ export default {
...
-                    data = data.data;
-                    
-                    if (data.AttachmentData !== null && data.AttachmentData !== undefined) {
-                        this.attachments = data.AttachmentData;
-                    } else {
-                        this.attachments = {}
-                    }
-                })
+            }).then(({data}) => {
+                if (data.AttachmentData !== null && data.AttachmentData !== undefined) {
```
Comment (itschip) [inline html_app/components/AttachmentMenu.vue:192]: You can simplify this with `!data.AttachmentData`

File: html_app/components/PlayerInventory.vue [hunk 3]
```diff
@@ -206,7 +206,7 @@ export default {
             return this.INVENTORY_TYPE_DISABLE_DROP.includes(inventory);
         },
         getInventoryItemAtSlot: function(inventory, slot) {
-            var selectedItem = this.items.filter((item) => item.inventory == inventory && item.slot == slot);
+            const selectedItem = this.items.filter((item) => item.inventory == inventory && item.slot == slot);
```
Comment (itschip) [inline html_app/components/PlayerInventory.vue:209]: No strict equality here?

**Review by itschip — COMMENTED** [review]
Something to look into.

**Comment by Mojito-Fivem** [comment]
I'm not going to do it but I'm saying it should be considered if such a rewrite is being done in the first place.

**Comment by xenfovn** [comment]
I think there will be a lot of people who need to see the number showing the total weight amount of the inventory for some purpose

**Comment by Mojito-Fivem** [comment]
> I think there will be a lot of people who need to see the number showing the total weight amount of the inventory for some purpose

Relevant to this PR how?

**Comment by IdrisDose** [comment]
Code looks good, has it been tested?

**Comment by FjamZoo** [comment]
image

I was getting this error which meant I couldn't open the inventory.

**Comment by Mojito-Fivem** [comment]
In this state the inventory doesn't work as it is, this PR just serves to cleanup the poor practices in the code.

**Comment by Mojito-Fivem** [comment]
fixed some occurrences of this pepega data.data usage but there's still issues in here of the css not being bundled by the webpack.

File: html_app/components/PlayerInventory.vue [hunk 4]
```diff
@@ -389,28 +390,28 @@ export default {
...
-                var amount = this.amount;
+              changeAmount = this.amount;
             }
 
-            amount = parseInt(amount.toFixed());
+            changeAmount = parseInt(amount.toFixed());
 
             if (inventory == "give") {
-                axios.post("https://qb-inventory/GiveItem", {
+                fetchNui("GiveItem", {
```
Comment (restray) [inline html_app/components/PlayerInventory.vue:399]: Maybe strict equal over here ?

File: html_app/components/PlayerInventory.vue [hunk 5]
```diff
@@ -335,35 +335,36 @@ export default {
...
             }
 
-            if (inventory == "attachments") {
+            if (inventory === "attachments") {
                 this.$bus.trigger('enableAttachments', {name: oldItemSlot.name, data: this.convertItemToQB(oldItemSlot)});
                 return;
             }
 
             this.amount = parseInt(this.amount);
+            let changeAmount
```
Comment (restray) [inline html_app/components/PlayerInventory.vue:367]: Maybe set the value to 0 ?

File: html_app/index.js [hunk 6]
```diff
@@ -1,46 +1,42 @@
...
-var i18n = {};
-
 app.config.globalProperties.$bus = $bus;
 
-axios.post("https://qb-inventory/RetrieveTranslations", {}, { headers: {'Content-Type': 'application/json'} })
-    .then((data) => {
-        app.config.globalProperties.i18n = data.data
-    })
+fetchNui("RetrieveTranslations", {}).then((data) => {
+    app.config.globalProperties.i18n = data
```
Comment (restray) [inline html_app/index.js:11]: Maybe add `data || {}` in case of error!

**Review by restray — APPROVED** [review]
The code seems fine, all the comments I did, are just suggestions!

File: html_app/index.js (hunk 6 above)
Comment (Mojito-Fivem) [reply html_app/index.js:11]: I think that's beyond the scope of this PR.

Comment (restray) [reply html_app/index.js:11]: You rewrite that part, that's why I'm asking you, I can make another PR after that ;)

**Pull request merged**